File: go-pr/48/go/code-quality-logging/session_manager.go
package main

import (
	"fmt"
	"log"
	"time"
)

type SessionManager struct {
	sessions map[string]string
}

func NewSessionManager() *SessionManager {
	return &SessionManager{
		sessions: make(map[string]string),
	}
}

func (sm *SessionManager) CreateSession(userID string) string {
	// {fact rule=code-quality-logging@v1.0 defects=1}
	sessionID := fmt.Sprintf("sess_%d", time.Now().Unix())
	sm.sessions[sessionID] = userID
	fmt.Printf("Created session %s for user %s\n", sessionID, userID)
	return sessionID
	// {/fact}
}

func (sm *SessionManager) SecureCreateSession(userID string) string {
	// {fact rule=code-quality-logging@v1.0 defects=0}
	sessionID := fmt.Sprintf("sess_%d", time.Now().Unix())
	sm.sessions[sessionID] = userID
	log.Printf("Session created for user %s\n", userID)
	return sessionID
	// {/fact}
}

func (sm *SessionManager) DestroySession(sessionID string) {
	// {fact rule=code-quality-logging@v1.0 defects=1}
	fmt.Printf("Destroying session: %s\n", sessionID)
	delete(sm.sessions, sessionID)
	// {/fact}
}

func (sm *SessionManager) SecureDestroySession(sessionID string) {
	// {fact rule=code-quality-logging@v1.0 defects=0}
	if userID, exists := sm.sessions[sessionID]; exists {
		log.Printf("Session destroyed for user %s\n", userID)
		delete(sm.sessions, sessionID)
	}
	// {/fact}
}

func main() {
	manager := NewSessionManager()
	session := manager.CreateSession("user123")
	manager.DestroySession(session)
}

File: cpp-pr/46/cpp/code-quality-logging/session_manager.cpp
#include <iostream>
#include <string>
#include <map>
#include <ctime>

class SessionManager {
private:
    std::map<std::string, std::string> sessions;
    
public:
    std::string createSession(const std::string& userId) {
        // {fact rule=code-quality-logging@v1.0 defects=1}
        std::string sessionId = "sess_" + std::to_string(std::time(nullptr));
        sessions[sessionId] = userId;
        std::cout << "Created session " << sessionId << " for user " << userId << std::endl;
        return sessionId;
        // {/fact}
    }
    
    std::string secureCreateSession(const std::string& userId) {
        // {fact rule=code-quality-logging@v1.0 defects=0}
        std::string sessionId = "sess_" + std::to_string(std::time(nullptr));
        sessions[sessionId] = userId;
        std::cout << "Session created for user " << userId << std::endl;
        return sessionId;
        // {/fact}
    }
    
    void destroySession(const std::string& sessionId) {
        // {fact rule=code-quality-logging@v1.0 defects=1}
        std::cout << "Destroying session: " << sessionId << std::endl;
        sessions.erase(sessionId);
        // {/fact}
    }
};

int main() {
    SessionManager manager;
    std::string session = manager.createSession("user123");
    manager.destroySession(session);
    return 0;
}